### PR TITLE
Cleanup of generateVAO() to remove special casing of TextureShader. Reuse VAO and VBOids and make generateVAO() private.

### PR DESCRIPTION
**Pull request by Vittalbabu**
Discussed with @roshanch the following cleanup of generateVAO() usage. Tested shadow, jassimp, simplescene, litshader, multilight samples.

Remove the special case for TextureShader to call generateVAO(). Now generateVAO() will be called just before the glBindVertexArray(mesh->getVAOId(programId)) call and after the shader->render() call.

TextureShader's render method sets the proper program instance for the light and no_light case and the programId will be valid after the call to render().

### CODE REVIEW

File: GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp [hunk 1]
```diff
@@ -772,6 +767,11 @@ void Renderer::renderMaterialShader(RenderState& rstate, RenderData* render_data
                 error.c_str());
         shader_manager->getErrorShader()->render(&rstate, render_data, curr_material);
     }
+    programId = shader->getProgramId();
+    if (programId != -1)
```
Comment (roshanch) [inline GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp]: what would be the case when programId is -1 ?

Comment (Vittalbabu) [inline GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp]: It is a sanity check. This case should not occur. I can leave this as is or trigger an error in generateVAO() for invalid programId.

**Comment by Vittalbabu** [comment]
1. Reuse VAO and VBO ids when attributes change in mesh.cpp
2. Mesh::getVAOId() will now invoke generateVAO() as needed -- there is no need for the Renderer class to invoke this method. generateVAO() is now private.
3. Added method to delete VAO and VBO ids for a particular programId. This is not used for now.

**Comment by roshanch** [comment]
i think title is bit misleading. its more about deleting existing VBO and VAO ids when we the mesh changes. Code looks good to me. Needs to test it.

**Comment by liaxim** [comment]
Needs DCO

**Comment by Vittalbabu** [comment]
GearVRf-DCO-1.0-Signed-off-by: Vittalbabu Srinivasan
vittalbab.s@samsung.com

**Comment by Vittalbabu** [comment]
Tested with Shadows, jassimp, litshader and a modified version of polyline where a click changes the vertex data of the polyline.

**Comment by liaxim** [comment]
Will merge in the afternoon

**Pull request merged**